### PR TITLE
feat: load by parent category

### PR DESCRIPTION
**Pull request by nicolocarpignoli**
Load categories by parent category id if given

### CODE REVIEW

File: src/Controller/Component/CategoriesComponent.php [hunk 1]
```diff
@@ -37,12 +37,24 @@ public function initialize(array $config)
      * Load categories by object type.
      *
      * @param string|null $type Object type name.
+     * @param string|null $parentId ID of parent category.
      * @return \Cake\ORM\Query
      */
-    public function load(?string $type = null): Query
+    public function load(?string $type = null, ?string $parentId): Query
```
Comment (fquffio) [inline src/Controller/Component/CategoriesComponent.php:43]: `$parentId` should be of type `int`

File: src/Controller/Component/CategoriesComponent.php [hunk 2]
```diff
@@ -37,12 +37,24 @@ public function initialize(array $config)
...
+            ->where([$this->Categories->aliasField('enabled') => true]);
+
+        if ($parentId !== null) {
+            $query = $query
+                ->where([$this->Categories->aliasField('parent_id') => $parentId]);
+        }
+
+        if ($type !== null) {
+            $query = $query->find('type', [$type]);
+        }
```
Comment (fquffio) [inline src/Controller/Component/CategoriesComponent.php:55]: Query objects are lazily executed, so it was perfectly fine to add `where()` conditions and append finders after setting `order()` and results formatter. However, if deemed clearer, it's perfectly fine to group conditions first. 👍🏼

**Review by fquffio — CHANGES_REQUESTED** [review]

File: src/Controller/Component/CategoriesComponent.php (hunk 2 above)
Comment (nicolocarpignoli) [reply src/Controller/Component/CategoriesComponent.php:55]: It seems clearer to me to first filter by given filters, if any.

File: src/Controller/Component/CategoriesComponent.php (hunk 1 above)
Comment (nicolocarpignoli) [reply src/Controller/Component/CategoriesComponent.php:43]: correct, fixed.

**Pull request merged**